### PR TITLE
:wrench: set sources Compatibility to Java 11

### PR DESCRIPTION
**Pull request by etiennecadicidean**
For multi-module projects, the gradle 7+ ready version was raising ambigus variant selection for modules resolutions. 

Thanks to this thread : https://github.com/gradle/gradle/issues/12126 I was able to understand the issue was with a wrong sourceCompability attribute set in the jk1 plugin. 

It was indeed still declaring 
```groovy
sourceCompatibility = 1.8
targetCompatibility = 1.8
```
instead of 

```groovy
sourceCompatibility = JavaVersion.VERSION_11
targetCompatibility = JavaVersion.VERSION_11
```
though it was declared to be "gradle 7+ ready".

Changing to JavaVersion.VERSION_11 and building the plugin in local solved the issue from my end.

### CODE REVIEW

**Comment by jk1** [comment]
First of all, thanks for the contribution. All this variant-attribute-capability story adds a whole new build complexity dimension to things that are complex enough. Android builds were always a hard stuff to handle, so I'm not exactly surprised something doesn't work there.

I cannot accept the PR as it is, since it goes as far as replacing visible license names with their SPDX identifiers. Adding SPDX information is a great idea; spoiling what people see in reports - not that much.

**Comment by etiennecadicidean** [comment]
Hi @jk1 , sorry for the 2 commits. I added the 2nd one for our sonar plugin to be able to match with the jk1 report. At this point I completely forgot I had a PR opened on master (was 6 days after I created the PR). 

Will open another PR just for the 1st commit as this should definitely help lots of users.

I agree thta supporting the SPDX identifiers should be another distinct feature with much more impacts on the whole project. Had to to it as a quick win for our needs though.

**Comment by jk1** [comment]
@etiennecadicidean I've already extracted and applied the first commit, as it makes total sense.
Please try 2.1 version out and let me know if it works for your project. I hope it does.

**Comment by etiennecadicidean** [comment]
@jk1 thank you. 

Actually, I just tried to use 2.1 version and I came across a really weird issue that keeps driving me nuts :

I still have the issue in 2.1 version.
BUT
it has nothing to do with the change you made : it looks like there is something wrong with 'new' gradle plugin dsl : 

if I use the "old" way to declare the plugin : 

```Groovy
 repositories {
        maven {
            url "https://plugins.gradle.org/m2/"
        }
    }
///....
    dependencies {
        classpath 'com.github.jk1:gradle-license-report:2.0'
     }

```

and then in my module :

```Groovy
plugins {
    id "com.github.jk1.dependency-license-report"
}

```

it actually WORKS even with 2.0 version...

I thought I fixed the problem with changing the sourceset with JAVA_11 but in fact, I fixed the problem because in order to use local Jar, I had to use the old DSL way to declare plugin, with classpath (files(....))


to sum up : 

using

```Groovy
plugins {
    id "com.github.jk1.dependency-license-report" version "2.X"
}
```

fails in multi-module projects with gradle 7+, and it fact, it produces the very same issue than the one I could get using '1.X' version

while using the good old classpath definition works; even for 2.0 version.

I don't think the change you made was wrong at all but the issue I'm facing leads me to think there is a bug somewhere in gradle, because it looks like the version is actually ignored and we get an old plugin not compatible with gradle 7 and java 11...
